lib/iris/fileformats/_ff.py

```diff
@@ -717,76 +717,85 @@ def _extract_field(self):
+
+                    field.x, field.y = grid.vectors(subgrid)
+
+                    # Use the per-file grid if no per-field metadata is
+                    # available.
+                    no_x = field.bzx in (0, field.bmdi) and field.x is None
+                    no_y = field.bzy in (0, field.bmdi) and field.y is None
+                    if no_x and no_y:
+                        if subgrid is None:
+                            msg = ('The STASH code {0} was not found in the '
+                                   'STASH to grid type mapping. Picking the P '
+                                   'position as the cell type'.format(stash))
+                            warnings.warn(msg)
+                        field.bzx, field.bdx = grid.regular_x(subgrid)
+                        field.bzy, field.bdy = grid.regular_y(subgrid)
+                        field.bplat = grid.pole_lat
+                        field.bplon = grid.pole_lon
+                    elif no_x or no_y:
+                        warnings.warn(
+                            'Partially missing X or Y coordinate values.')
+
+                    # Check for LBC fields.
+                    is_boundary_packed = self._ff_header.dataset_type == 5
+                    if is_boundary_packed:
+                        # Apply adjustments specific to LBC data.
+                        self._adjust_field_for_lbc(field)
+
+                    # Calculate start address of the associated PP header data.
+                    data_offset = field.lbegin * self._word_depth
+                    # Determine PP field payload depth and type.
+                    data_depth, data_type = self._payload(field)
+
+                    # Produce (yield) output fields.
+                    if is_boundary_packed:
+                        fields = self._fields_over_all_levels(field)
+                    else:
+                        fields = [field]
+                    for result_field in fields:
+                        # Add a field data element.
+                        if self._read_data:
+                            # Read the actual bytes. This can then be converted
+                            # to a numpy array at a higher level.
+                            ff_file_seek(data_offset, os.SEEK_SET)
+                            result_field._data = pp.LoadedArrayBytes(
+                                ff_file.read(data_depth), data_type)
+                        else:
+                            # Provide enough context to read the data bytes
+                            # later.
+                            result_field._data = (self._filename, data_offset,
+                                                  data_depth, data_type)
 
-                    data_offset += data_depth
+                        data_offset += data_depth
 
-                    yield result_field
+                        yield result_field
+                except ValueError as valerr:
```

Comment:
I think the code here is a bit too coupled to the pp module implementation.
It has the following problems :
 * we are *assuming* here that any ValueError must be an LBREL match, but that could easily change in the pp module  -- as we are creating a new field, the ```PPField.__init__``` call could easily add some other possible failure mode.
 * we are throwing away the detail of the provided message, so the user does not see the intended information about the reason (the LBREL value it didn't like).

I think it is a better idea to defer the *reason* for failure and the explanation to the PP module itself and just say here something like :
```
    msg = "Input field skipped because PPField creation failed : error = {!r}"
    raise ValueError(msg.format(str(valerr)))
```